app/code/core/Mage/Catalog/Model/Product/Type/Configurable.php

```diff
@@ -324,7 +325,8 @@ public function getUsedProducts($requiredAttributeIds = null, $product = null)
         Varien_Profiler::start('CONFIGURABLE:' . __METHOD__);
         if (!$this->getProduct($product)->hasData($this->_usedProducts)) {
             if (is_null($requiredAttributeIds)
-                and is_null($this->getProduct($product)->getData($this->_configurableAttributes))) {
+                and is_null($this->getProduct($product)->getData($this->_configurableAttributes))
```

Comment:
Oh no a `and` instead of '&&'. Ok this is not the goal of the PR.

Reply:
Good catch - for next PR :)

Reply:
Wait for next PR.